dali/test/python/test_operator_numpy_reader.py

```diff
@@ -122,3 +216,62 @@ def check_array(filename, shape, typ, fortran_order=False):
 
     # delete temp files
     delete_numpy_file(filename)
+
+
+def check_array_static_slab(filename, shape, slab_anchor, slab_shape, typ, io_size=0, fortran_order=False):
+    # setup file
+    create_numpy_file(filename, shape, typ, fortran_order)
+    
+    for num_threads in [1, 2, 4, 8]:
+	# load with numpy reader
+        pipe = NumpyReaderPipeline(path = os.path.dirname(filename),
+                                   path_filter = os.path.basename(filename),
+                                   batch_size = 1,
+                                   num_threads = num_threads,
+                                   device_id = 0,
+                                   slab_anchor = slab_anchor,
+                                   slab_shape = slab_shape,
+                                   io_size = io_size)
+        pipe.build()
+        pipe_out = pipe.run()
+        arr_rd = np.squeeze(pipe_out[0].as_array(), axis=0)
+
+        # load manually
+        arr_np = np.load(filename)
+        slab = [slice(x[0], x[0]+x[1]) for x in zip(slab_anchor, slab_shape)]
+
+        print(arr_rd, arr_np)
+        
+        # compare
+        assert_array_equal(arr_rd, arr_np[slab])
+
+    # delete temp files
+    delete_numpy_file(filename)
+
+
+def check_array_dynamic_slab(filename, shape, typ, fortran_order=False):
+    # setup file
+    create_numpy_file(filename, shape, typ, fortran_order)
+
+    for num_threads in [1, 2, 4, 8]:
+        # load with numpy reader
+        pipe = NumpyReaderDynamicPipeline(path = os.path.dirname(filename),
+                                          path_filter = os.path.basename(filename),
+                                          batch_size = 1,
+                                          num_threads = num_threads,
+                                          device_id = 0)
+        pipe.build()
```

Comment:
Maybe you can compare Numpy(anchor/shape) and Numpy+slice(anchor/shape).

Reply:
I have tests which compare loading the whole file into a numpy array using np.load and then slicing to the numpy sliced loader. Since I am using some contiguous dim fusing, I am also using slices which have fusable dims. Or what else to you mean?

Reply:
Ok. I see now. You get the anchors and shapes from the pipeline so that should do.